### ISSUE TITLE
Impossible to play back sounds that start with immeidate, sharp transients

### ISSUE THREAD

**Issue by wareya** [audacity/audacity#5084]
### Bug description

When playing back sounds that have a sharp transient with the cursor exactly at or before the transient, the transient is skipped during playback, as if it's being faded-in over. This makes audacity extremely difficult to use for video game sound effect design, where immediate transients without any left-pad silence are often vital to tight gamefeel.

Micro-fades are disabled in Audacity's preferences.

### Steps to reproduce

1) Create or acquire a sound with a sharp transient, that starts at exactly the transient, without any left silence padding.
2) Place the cursor at or immediately before (on the scale of individual wave samples) the transient.
3) Hit play.
4) Observe the lack of transient hit.

### Expected behavior

The sharp transient should be audible.

### Actual behavior

The sharp transient is not audible.

### Audacity Version

latest stable version (from audacityteam.org/download)

### Operating system

Windows

### Additional context

https://github.com/audacity/audacity/assets/585488/29c590d1-f63a-4a8c-858b-dfdd2045fa66

(unmute)

[pop.wav.zip](https://github.com/audacity/audacity/files/12433685/pop.wav.zip)

**Comment by LWinterberg** [comment]
This issue seems to happen in WASAPI only. MME, ASIO and DirectSound work correctly.

**Comment by im-made-of-jam** [comment]
I realise this is a very old thread, but this appears to be a WASAPI issue, not an Audacity issue, at least for Windows 11. I have come across this issue recently when using WSAPI with nothing other than the C++ standard library and the windows headers, so I would be inclined to say this is a Microsoft problem instead.

**Comment by wareya** [comment]
No, if this is WASAPI's "fault" so to speak then that means that whatever is talking to it is talking to it wrong. For example, not generating leading silence when going from not talking to WASAPI to yes talking to WASAPI.

**Comment by im-made-of-jam** [comment]
I would classify WASAPI having a built in fade in as its own issue.
That said, my workaround for this was to simply initialise the audio device earlier, and only start it after a short time so that the fade in was avoided.
I am not familiar enough with Audacity to implement a workaround myself but I would guess initialising the audio device sooner then only playing audio through it after a short time would work to alleviate this issue.
